backend-api/src/functions/asyncActiveSession/tokenService/tokenService.ts

```diff
@@ -78,6 +97,47 @@ export class TokenService implements ITokenService {
     );
   };
 
+  private async getJwks(
+    stsJwksEndpoint: string,
+    retryConfig: RetryConfig,
+  ): Promise<Result<IJwks>> {
+    const sendHttpRequestResult = await sendHttpRequest(
+      { url: stsJwksEndpoint, method: "GET" },
+      retryConfig,
+    );
+
+    if (sendHttpRequestResult.isError) {
+      return sendHttpRequestResult;
+    }
+
+    const jwksEndpointResponse = sendHttpRequestResult.value;
+
+    const getJwksFromResponseResult =
+      await this.getJwksFromResponse(jwksEndpointResponse);
+
+    if (getJwksFromResponseResult.isError) {
+      return getJwksFromResponseResult;
+    }
+
+    const jwks = getJwksFromResponseResult.value;
+
+    return successResult(jwks);
+  }
+
+  private getJweComponents(jwe: string): Result<string[]> {
+    const jweComponents = jwe.split(".");
+
+    if (jweComponents.length !== 5) {
+      return errorResult({
+        errorMessage:
+          "Decrypt service token failure: JWE does not consist of five components",
```

Comment:
This implies to me we have a decrypt service but I don't think we do?

Soz commented on the test version of this first

Reply:
I see how it can be read in two way. I've remove the first part of the error message to avoid ambiguity